synapse/handlers/federation.py

```diff
@@ -2611,16 +2611,6 @@ async def _check_event_auth(
             auth_events_x = await self.store.get_events(auth_events_ids)
             auth_events = {(e.type, e.state_key): e for e in auth_events_x.values()}
 
-        # This is a hack to fix some old rooms where the initial join event
-        # didn't reference the create event in its auth events.
-        if event.type == EventTypes.Member and not event.auth_event_ids():
-            if len(event.prev_event_ids()) == 1 and event.depth < 5:
-                c = await self.store.get_event(
-                    event.prev_event_ids()[0], allow_none=True
-                )
-                if c and c.type == EventTypes.Create:
-                    auth_events[(c.type, c.state_key)] = c
```

Comment:
What's the reasoning for not needing this anymore? Such old rooms may still be floating around? Possibly an alternative is to only do this for v1 rooms?

Reply:
This workaround was added in https://github.com/matrix-org/synapse/commit/858e87ab0d1bd88fcd5c819c482a40130e571874, from which I infer it was fixed before then. This is only a week after ae9c2ab1652511b36dc594882e88152b6067c816, which we know was a critical security fix. I'm pretty sure we previously checked that there weren't any remaining rooms affected by the SYN-149 bug, but if there are any other rooms that predate that fix... I don't think I want to join them.

Reply:
Hmm, OK fair